framework/tlvf/yaml/tlvf/wfa_map/tlvClientCapabilityReport.yaml

```diff
@@ -10,12 +10,9 @@ tlvClientCapabilityReport:
     _value_const: TLV_CLIENT_CAPABILITY_REPORT
   length: uint16_t
   result_code: eResultCode
-  association_frame_length:
-    _type: uint8_t
-    _length_var: True
   association_frame:
-    _type: uint8_t
-    _length: [ association_frame_length ] 
+    _type: char
```

Comment:
> so we can easilly print the association frame using `association_frame_str()`

The association frame is binary data. Printing it as a string will not lead to much joy.

I believe that what you parse out of the AP-STA-CONNECTED event from hostapd is the hex representation of the frame. So it should be converted back to binary (uint8_t) inside bwl and passed as binary in the Client Capability Report TLV.

As Tomer says, check with a certified device what the contents is, and make sure we generate the same thing. Actually, it would be nice if you could upload a pcap file with this TLV as generated by a certified device, so everybody can easily compare with it.